fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/FedoraLdpIT.java

```diff
@@ -2525,6 +2524,38 @@ public void testPathsWithBrackets() {
     }
 
     @Test
+    public void testConcurrentPuts() throws InterruptedException, IOException {
+        final String parent = getRandomUniqueId();
+        executeAndClose(putObjMethod(parent));
+        final String newResource = parent + "/test";
+        final Runnable updateRunnable = () -> { executeAndClose(putObjMethod(newResource)); };
+        final Thread t1 = new Thread(updateRunnable);
+        final Thread t2 = new Thread(updateRunnable);
+        final Thread t3 = new Thread(updateRunnable);
+        final Thread t4 = new Thread(updateRunnable);
+        t1.start();
+        t2.start();
+        t3.start();
+        t4.start();
+        t1.join();
+        t2.join();
+        t3.join();
+        t4.join();
+
+        try (final CloseableDataset dataset = getDataset(getObjMethod(parent))) {
+            final DatasetGraph graphStore = dataset.asDatasetGraph();
+            final Iterator<Quad> children = graphStore.find(ANY, ANY, CONTAINS.asNode(), ANY);
+            assertTrue("One of the PUTs should have resulted in a child.", children.hasNext());
+            children.next();
+            if (children.hasNext()) {
```

Comment:
Okay, I see that. This (lazy messages) is solved in [JUnit 5](http://junit.org/junit5/docs/current/api/org/junit/jupiter/api/Assertions.html#assertFalse-boolean-java.util.function.Supplier-), but that'll be a while to come.